lib/model.js

```diff
@@ -827,6 +827,27 @@ ModelBaseClass.getMergePolicy = function(options) {
   return mergePolicy;
 };
 
+/**
+ * Gets properties defined with 'updateOnly' flag set to true from the model. This flag is also set to true
+ * internally for the id property, if this property is generated and IdInjection is true.
+ * @returns {updateOnlyProps} List of properties with updateOnly set to true.
+ */
+
+ModelBaseClass.getUpdateOnlyProperties = function() {
+  var Model = this;
+  var props = Model.definition.properties;
+  var property;
+  var updateOnlyProps = [];
+
+  for (var key in props) {
```

Comment:
Let's use [Array.prototype.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to simplify this code.

```js
const props = this.definition.properties;
return Object.key(props).filter(key => props[key].updateOnly);
```